Sources/StreamChat/Database/DTOs/ChannelReadDTO.swift

```diff
@@ -63,19 +69,78 @@ class ChannelReadDTO: NSManagedObject {
 // MARK: Saving and loading the data
 
 extension NSManagedObjectContext {
+    func upsertMany(payload: [ChannelReadPayload], for cid: ChannelId) throws -> [ChannelReadDTO] {
```

Comment:
would be nice to have a generic upsert function that is called from these specific implementations

Reply:
I agree, once we are happy with the approach we can do that. Not going to do that until we are 100% sold on this approach (making this generic is not trivial)